pkg/services/libraryelements/database.go

```diff
@@ -92,10 +92,17 @@ func (l *LibraryElementService) createLibraryElement(c *models.ReqContext, cmd C
 	if err := l.requireSupportedElementKind(cmd.Kind); err != nil {
 		return LibraryElementDTO{}, err
 	}
+	UID := cmd.UID
+	if len(UID) == 0 {
+		UID = util.GenerateShortUID()
+	}
+	if !util.IsValidShortUID(UID) {
```

Comment:
You probably want to validate the length as well, as can be seen here:
https://github.com/grafana/grafana/blob/84e821bdeffda0d25d4c20c5256d7c7b7f3ec126/pkg/services/dashboards/dashboard_service.go#L104-L108

Reply:
Ok, wouldn't it make sense to add that check to the `IsValidShortUID` function instead?

Reply:
Or maybe add a new function called IsValidShortUIDLength?

Reply:
Created an `IsShortUIDTooLong` function and tests.